### ISSUE TITLE
Revert to the old merge conflict view

### ISSUE THREAD

**Issue by scooby359** [microsoft/vscode#157610]
Type: <b>Feature Request</b>

I find the new conflict view really difficult to use. Harder for me to see what changes there are in the side by side view, and not clear how I'm supposed to accept changes from one or the other - sometimes it lets me tick a box at the side, other times it doesn't. Doesn't make sense. Let me go back to the old view!

VS Code version: Code 1.70.0 (da76f93349a72022ca4670c1b84860304616aaa2, 2022-08-04T04:38:16.462Z)
OS version: Windows_NT x64 10.0.22000
Modes:

**Comment by nitishxp** [comment]
@scooby359  I agree , it caused a nightmare to resolve a single line how to get back to older version

**Comment by shashankshawak7** [comment]
Please revert it to the old version , or solve my merge conflict for me :P

**Comment by DutchGerman** [comment]
You can disable the new merge editor by adding the following to you settings.json

```json
{
  "git.mergeEditor": false
}
```

**Comment by flexbox** [comment]
Same for me...

Dear Microsoft, realasing a merge diff tool on Friday without opt-in is a terrible idea

**Comment by xander-marjoram** [comment]
The feature that I use most often is "Accept both changes", then manually fix.
It's not obvious how to do this with the new merge editor.

**Comment by IllusionMH** [comment]
> The feature that I use most often is "Accept both changes", then manually fix.
> It's not obvious how to do this with the new merge editor.

Same as in other 3-way editors - accept both (set checkboxes in left and right editors) and then update result view

**Comment by NickCraver** [comment]
For what it's worth, I found myself in the new merge editor today and could not figure out how to actually merge anything intuitively. Right clicking does nothing, dragging does nothing, I still don't really know what you're supposed to do with it, and a view split between 3 editors even on a 27" monitor is very cramped. Given I usually click "Accept X" or "Accept Both" for each merge quickly, this was quite a regression in productivity (though maybe it wouldn't be if I understood how you're supposed to use it). Only after finding this issue did I notice the checkboxes at all.

I just wanted to chime in that this default change was very unintuitive and a net negative for me - after a few minutes trying to figure out, I gave up and went to find how to flip it off in settings.

> Same as in other 3-way editors - accept both (set checkboxes in left and right editors) and then update result view

@IllusionMH Even after reading and trying this, I don't know how a user is supposed to figure that out. If I check the left checkbox, the left line appears in the result view below. If I check the right box, the right line appears in the result below. So far so good. If I check both **neither** line appears in the result below. I have no idea what to make of that, but it's either bugged or very unintuitive and I need an explainer video or something.

To illustrate what I mean:

Both unchecked: both lines appear in result (not expected)
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/454813/183667225-adfe16df-3b4a-4c85-a459-7fa0564eab87.png">
Left checked: left line appears in result (expected)
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/454813/183667432-d2de7aa7-5c0d-4646-8e99-045342afb2b1.png">
Right checked: right line appears in result (expected)
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/454813/183667506-eae8b63f-f256-4d67-8e1b-14415712c7b2.png">
Both checked: neither line appears in result (unexpected)
<img width="1148" alt="image" src="https://user-images.githubusercontent.com/454813/183667608-c9da4197-a63a-4488-82d2-47687bf490aa.png">

**Comment by schemburkar** [comment]
Agree with the comments from @NickCraver and @scooby359.
We were left struggling to find which side to choose in the merge today. And the navigation buttons to move between conflicts was not working.

Whats worse is that if conflict is not resolved, it does not add the separation likes (====) in code so if by chance you select continue with conflicts, the file is left in a hybrid state with changes from some side (really not sure why) and difficult to get back. Unless you want to discard all the work and start again.


I would like to revert to the old way or at the very least have the option to opt in before making this the default.

**Comment by IllusionMH** [comment]
> So far so good. If I check both neither line appears in the result below.

Actually looks like a bug. @NickCraver , please, report this case separately.

---

As for default change - it was expected change announced when feature was introduced https://code.visualstudio.com/updates/v1_69#_3-way-merge-editor
As with other changes - shouldn't be a problem if there is clear way how to get previous behavior. And in general this means that more people will get this feature, use and like it.

IMHO in this case (as with some previous) change of default is not in Release notes so it requires go to previous version to find what setting to change to get old behavior.

---

Everyone who finds new editor confusing or have problems - please upvote existing issue and create new for cases that aren't covered so dev team would be able to address them.

**Comment by NickCraver** [comment]
@IllusionMH I gave I thought a pretty decent report here on the relevant issue, if someone wants to separate it out and track separately that's fine with me :) No need to tag even because I have no plans to use the new editor - it's just very cramped regardless of if the behavior's fixed.

I sincerely hope this default is reverted still - I use VSCode as a merge editor _because it's not a cramped 3-way merge editor_. It's my go-to on merges specifically _because_ of the previous default inline merge view.

**Comment by hediet** [comment]
@NickCraver and @schemburkar thanks for your feedback!

> I just wanted to chime in that this default change was very unintuitive and a net negative for me - after a few minutes trying to figure out, I gave up and went to find how to flip it off in settings.

Please note that this feature was not enabled by default for at least two months, in which everyone on our team used the feature.
Initial designs of this feature were discussed [here](https://github.com/microsoft/vscode/issues/146091) back in March. Only by making this feature default for everyone we got feedback like yours that we can use to improve the feature for everyone.

> Even after reading and trying this, I don't know how a user is supposed to figure that out. If I check the left checkbox, the left line appears in the result view below. If I check the right box, the right line appears in the result below. So far so good. If I check both neither line appears in the result below. I have no idea what to make of that, but it's either bugged or very unintuitive and I need an explainer video or something.

When you check no checkbox, you see the base version in the result.
In this case the base has two package entries. On both branches, one package item got deleted.
By checking the left/right checkbox you accept the deletion from the left/right branch.
Thus, when checking both checkboxes, both sides are deleted.

I agree that the presentation of the deletion is not great.
Can you invoke the command "Copy Merge Editor State As Json" and share the result here?

> IMHO in this case (as with some previous) change of default is not in Release notes so it requires go to previous version to find what setting to change to get old behavior.

We included it in the release notes:
https://code.visualstudio.com/updates/v1_70#_3way-merge-editor-improvements

But I agree, we could have more explicit about how to disable it.

**Comment by DutchGerman** [comment]
@hediet thanks for taking the time to answer, since this issue is about making the "inline" merge conflict view default again I have a question.

Is there a reason why we can't enable both at the same time? I can see the benefits of having a stronger merge editor for bigger merge conflicts, but I love the usability and simplicity of the old one for the smaller ones. Especially in a rebase scenario, where I rarely open the version control tab.

![image](https://user-images.githubusercontent.com/60694691/183702091-fa75cf24-1ea4-4b5c-9b55-855f7c30419c.png)

**Comment by NickCraver** [comment]
@hediet due to being an internal project, I can't share the JSON here (will need to wait for next merge anyway), but happy to via Teams. I definitely appreciate being default brings a lot of feedback you don't otherwise get, but I also hope decisions aren't set in stone or at the very least that the old mode is sticking around with no plans to remove. The old functionality is the #1 reason I use VS Code.

I get the explanation of checkboxes being deleted but I'd still argue that's counterintuitive to pretty much every checkbox out there which is "include this". If the lines were red or something then it may be obvious what's happening. I simultaneously appreciate all the effort put into the merge editor, while still thinking: maybe it isn't quite ready to be a default yet. It should be at least as intuitive as the previous version first, IMO. The previous version was 1) very simple and 2) very intuitive - it literally had clear text labels to click. While supporting more cases, the new editor is a large regression in those regards and that should carry quite a bit of weight.

**Comment by gnowland** [comment]
This is functionally a duplicate of #156608 and should be closed.

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! We figured it's covering the same as another one we already have. Thus, we closed this one as a duplicate. You can search for [similar existing issues](https://github.com/microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3A%22merge-editor%22). See also our [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Happy Coding!

**Comment by feeedback** [comment]
> The feature that I use most often is "Accept both changes", then manually fix.
> It's not obvious how to do this with the new merge editor.

It's not obvious how to do ANYTHING with the new merge editor.

**Comment by isidorn** [comment]
Thanks for all the feedback. This issue has our latest thinking https://github.com/microsoft/vscode/issues/160806
